network/hybrid/network.go

```diff
@@ -0,0 +1,274 @@
+			Location: to.StringPtr(helpers.Location()),
+			SecurityGroupPropertiesFormat: &network.SecurityGroupPropertiesFormat{
+				SecurityRules: &[]network.SecurityRule{
+					{
+						Name: to.StringPtr("allow_ssh"),
+						SecurityRulePropertiesFormat: &network.SecurityRulePropertiesFormat{
+							Protocol:                 network.TCP,
+							SourceAddressPrefix:      to.StringPtr("0.0.0.0/0"),
+							SourcePortRange:          to.StringPtr("1-65535"),
+							DestinationAddressPrefix: to.StringPtr("0.0.0.0/0"),
+							DestinationPortRange:     to.StringPtr("22"),
+							Access:                   network.Allow,
+							Direction:                network.Inbound,
+							Priority:                 to.Int32Ptr(100),
+						},
+					},
+					{
+						Name: to.StringPtr("allow_https"),
+						SecurityRulePropertiesFormat: &network.SecurityRulePropertiesFormat{
+							Protocol:                 network.TCP,
+							SourceAddressPrefix:      to.StringPtr("0.0.0.0/0"),
+							SourcePortRange:          to.StringPtr("1-65535"),
+							DestinationAddressPrefix: to.StringPtr("0.0.0.0/0"),
+							DestinationPortRange:     to.StringPtr("443"),
+							Access:                   network.Allow,
+							Direction:                network.Inbound,
+							Priority:                 to.Int32Ptr(200),
+						},
+					},
+				},
+			},
+		},
+	)
+
+	if err != nil {
+		return nsg, fmt.Errorf(fmt.Sprintf(errorPrefix, resourceName, err))
+	}
+
+	err = future.WaitForCompletion(cntx, nsgClient.Client)
+	if err != nil {
+		return nsg, fmt.Errorf(fmt.Sprintf(errorPrefix, resourceName, fmt.Sprintf("cannot get nsg create or update future response: %v", err)))
+	}
+
+	return future.Result(nsgClient)
+}
+
+// CreatePublicIP creates a new public IP
+func CreatePublicIP(cntx context.Context, ipName string) (ip network.PublicIPAddress, err error) {
+	resourceName := "public IP"
+	environment, _ := azure.EnvironmentFromURL(helpers.ArmEndpoint())
+	ipClient := getIPClient(environment.ActiveDirectoryEndpoint, environment.TokenAudience)
+	future, err := ipClient.CreateOrUpdate(
+		cntx,
+		helpers.ResourceGroupName(),
+		ipName,
+		network.PublicIPAddress{
+			Name:     to.StringPtr(ipName),
+			Location: to.StringPtr(helpers.Location()),
+			PublicIPAddressPropertiesFormat: &network.PublicIPAddressPropertiesFormat{
+				PublicIPAllocationMethod: network.Static,
+			},
+		},
+	)
+
+	if err != nil {
+		return ip, fmt.Errorf(fmt.Sprintf(errorPrefix, resourceName, err))
+	}
+
+	err = future.WaitForCompletion(cntx, ipClient.Client)
+	if err != nil {
+		return ip, fmt.Errorf(fmt.Sprintf(errorPrefix, resourceName, fmt.Sprintf("cannot get public ip address create or update future response: %v", err)))
+	}
+	return future.Result(ipClient)
+}
+
+// CreateNetworkInterface creates a new network interface
+func CreateNetworkInterface(cntx context.Context, netInterfaceName, nsgName, vnetName, subnetName, ipName string) (nic network.Interface, err error) {
+	resourceName := "network interface"
```

Comment:
>resourceName := "network interface" [](start = 1, length = 35)

I dont think this var is needed at all/

Reply:
This is being used for logging purposes. An error prefix id defined  in line 22. errorPrefix gets resource name in each resource creation method.